ee/modules/030-cloud-provider-openstack/hooks/migrate/migrate_terraform_state.go

```diff
@@ -0,0 +1,190 @@
+				if err != nil {
+					return err
+				}
+				terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.type", i), OpenstackV3ResourceType)
+				if err != nil {
+					return err
+				}
+			}
+			for j, terraformDependency := range terraformResource.Get("instances.0.dependencies").Array() {
+				if strings.Contains(terraformDependency.String(), "openstack_blockstorage_volume_v2") {
+					input.LogEntry.Infof("Found dependency %s with old resource in resource = %s and type = %s. Modifying dependency.", terraformDependency.String(), terraformResource.Get("name").String(), terraformResource.Get("type").String())
+					terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.instances.0.dependencies.%d", i, j), strings.ReplaceAll(terraformDependency.String(), OpenstackV2ResourceType, OpenstackV3ResourceType))
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+
+		var newTerraformState bytes.Buffer
+		err = json.Indent(&newTerraformState, terraformStateRaw, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		secret.Data[TerraformStateDataKey] = newTerraformState.Bytes()
+		_, err = kubeCl.CoreV1().
+			Secrets(TerraformStateNamespace).
+			Update(context.TODO(), &secret, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		input.LogEntry.Infof("Migration process completed for Secret/%s/%s.", TerraformStateNamespace, secret.ObjectMeta.Name)
+	}
+
+	input.LogEntry.Infof("Terraform migrate hook has been finished.")
+	return nil
+}
+
+func isSecretBackupExists(backupSecretName string, namespace string, kubeCl k8s.Client, input *go_hook.HookInput) (bool, error) {
+	input.LogEntry.Debugf("Function isSecretBackupExists: Starting function with parameters: backupSecretName=%s; namespace=%s", backupSecretName, namespace)
+	_, err := kubeCl.CoreV1().
+		Secrets(namespace).
+		Get(context.TODO(), backupSecretName, metav1.GetOptions{})
+	input.LogEntry.Debugf("Function isSecretBackupExists: Get secret. err=%s", err)
+
+	if errors.IsNotFound(err) {
+		input.LogEntry.Debugf("Function isSecretBackupExists: errors.IsNotFound(err) = %t. secret \"%s\" not found'. Return false and nil", errors.IsNotFound(err), backupSecretName)
+		return false, nil
+	}
+
+	if err == nil {
+		input.LogEntry.Debugf("Function isSecretBackupExists: err == nil. Return true and nil")
+		return true, nil
+	}
+
+	input.LogEntry.Debugf("Function isSecretBackupExists: err !=nil and errors.IsNotFound(err) no true. Return false and err")
+	return false, err
+}
+
+func BackupSecret(backupSecretName string, secret v1.Secret, namespace string, kubeCl k8s.Client, input *go_hook.HookInput) bool {
+	nodeName := secret.ObjectMeta.Labels["node.deckhouse.io/node-name"]
+	nodeGroup := secret.ObjectMeta.Labels["node.deckhouse.io/node-group"]
+	// nodeName := strings.TrimPrefix(secret.ObjectMeta.Name, "d8-node-terraform-state-")
+
+	input.LogEntry.Infof("Starting backup for Secret/%s/%s", namespace, secret.ObjectMeta.Name)
```

Comment:
i think we should remove this log entry

Reply:
I've removed this line: `input.LogEntry.Infof("Starting backup for Secret/%s/%s", namespace, secret.ObjectMeta.Name)`